From 4f620eba03a17f8908c4f6f80706ae4a108383a8 Mon Sep 17 00:00:00 2001
From: Mario Schuknecht <mario.schuknecht@dresearch-fe.de>
Date: Mon, 18 Mar 2019 13:10:30 +0100
Subject: [PATCH] Close VPU decoder instance HYP-20302

Close VPU decoder instance even if there is no decoder context yet. This
ensures that a decoder that has not yet been fully created is released.

Signed-off-by: Mario Schuknecht <mario.schuknecht@dresearch-fe.de>
---
 src/vpu/decoder.c | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/src/vpu/decoder.c b/src/vpu/decoder.c
index 684ea24..c208d4e 100644
--- a/src/vpu/decoder.c
+++ b/src/vpu/decoder.c
@@ -1159,8 +1159,11 @@ static void gst_imx_vpu_decoder_close(GstImxVpuDecoder *vpu_decoder)
 
 static void gst_imx_vpu_decoder_close_and_clear_decoder_context(GstImxVpuDecoder *vpu_decoder)
 {
-	if (vpu_decoder->decoder_context == NULL)
+	if (vpu_decoder->decoder_context == NULL) {
+		GST_INFO_OBJECT(vpu_decoder, "Clearing decoder");
+		gst_imx_vpu_decoder_close(vpu_decoder);
 		return;
+	}
 
 	GST_INFO_OBJECT(vpu_decoder, "Clearing decoder context");
 
-- 
2.16.4

